### ISSUE TITLE
Crash on macOS Sierra 10.12.5

### ISSUE THREAD

**Issue by asbjornu** [schismtracker/schismtracker#109]
Launching `schismtracker` on macOS Sierra 10.12.5 crashes it with the signal `Trace/BPT trap: 5` on the command line and [this problem report](https://gist.github.com/asbjornu/b52121cb8b6459c7323006d744916569). `schismtracker --version` runs just fine (and reports the version number `20170628`), so we might want to find a better test to execute on Travis to detect crashes like these.

I've performed a fresh build of Schism Tracker, although the `Version` reported in the problem report is `20120105`. I assume that's related to #77.

The only intelligible portion of the problem report I can find is this:

> Detected over-release of a `CFTypeRef`

Ideas?

**Comment by jangler** [comment]
"Trace/BPT trap: 5" seems to indicate a dylib loading issue. What dylibs are you using and where are they located relative to your working directory?

**Comment by asbjornu** [comment]
@jangler: I'm sorry to have to say "I have no idea" to those questions. I've just got this `Schism Tracker.app` bundle that I downloaded somewhere, sometime, and I've attempted to update the `Contents/MacOS/schismtracker` executable within it.

**Comment by esaruoho** [comment]
tried launching schismtracker and got this

Fri Jul 07 18:49:21 ~ >> /Users/esaruoho/Downloads/schismtracker/schismtracker ; exit;
dyld: Library not loaded: /usr/local/opt/sdl/lib/libSDL-1.2.0.dylib
  Referenced from: /Users/esaruoho/Downloads/schismtracker/schismtracker
  Reason: image not found
Abort trap: 6
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

**Comment by asbjornu** [comment]
@jangler: Can you please help me figure out what dylibs I'm using and where they're located?

**Comment by jangler** [comment]
Hm, I'm looking at the crash report again and it says that thread 0 crashed, which was only using libsystem_kernel.dylib and libsystem_pthread.dylib, which are pretty fundamental things unrelated to any dylibs that the Schism app bundle would provide. So essentially, I have no leads. Doesn't help that I've never done any macOS dev.

**Comment by asbjornu** [comment]
Could the `.dylib`s bundled in the `.app` be outdated and incompatible with the `schismtracker` binary somehow? How can I rearrange the loaded `.dylib`s locally to figure out what might be awry? Could this be an error caused by how Schism Tracker is compiled and that the flags, paths or similar are off on macOS?

**Comment by asbjornu** [comment]
The crash report also say:

> Application Specific Information:
> Detected over-release of a `CFTypeRef`

So the problem seems to be related to these two stack frames:

0. `com.apple.CoreFoundation 0x00007fffcf94e802 _CFRelease + 1346`
1. `org.schismtracker.SchismTracker	0x0000000104d18df4 get_ep_name + 228`

Which leads me to believe that one of the following three calls to `CFRelease` within the `get_ep_name` function are to blame:

```c
/* clean up */
if (endpointName) CFRelease(endpointName);
if (deviceName) CFRelease(deviceName);
if (fullName) CFRelease(fullName);
```

From [this answer on Stack Overflow](https://stackoverflow.com/a/17270903/61818), I gather that you should only do `CFRelease` on objects you've created yourself with `CF…Copy…()` or `CF…Create…()`. Of the above variables, only `fullName` is created with `CFStringCreateWithFormat()`, and only in the situation where `deviceName` is `null`, so if it is set to the pointer of `deviceName`,  it should too not be released afaict.

So my basic understand is that there's a bug in Schism Tracker related to over-eagerly releasing objects that aren't Schism Tracker's to release since it hasn't created them in the first place. I'll try to adjust the source code and see if I can work out a fix, but with my knowledge in C I don't have high hopes. 😄

**Comment by agargara** [comment]
Confirmed same problem on OSX Sierra 10.12.6.

**Comment by asbjornu** [comment]
This is fixed in #116.

**Comment by agargara** [comment]
Thank you! I pulled your branch, compiled and ran. Seems to work without any problems now.

**Comment by jangler** [comment]
Forgot to close this!

**Comment by asbjornu** [comment]
@jangler: This would have been closed if you had merged #116. :smile:

**Comment by jangler** [comment]
Forgot to merge that!
